keyform/views.py

```diff
@@ -15,11 +17,55 @@ class HomeView(LoginRequiredMixin, ListView):
 
 class RequestView(LoginRequiredMixin, UpdateView):
     model = Request
+    success_url = reverse_lazy('home')
     template_name = "keyform/request.html"
     form_class = EditForm
 
     def get_success_url(self):
-        return reverse('home')
+        return reverse_lazy('home')
+
+class ContactView(TemplateView):
+    template_name = "keyform/contact.html"
+
+    def get_context_data(self):
+        context = super(ContactView, self).get_context_data()
+        context["buildings"] = Building.objects.all()
+        return context
+
+    def post(self, request, *args, **kwargs):
+        print(request.POST)
+
+        pk = request.POST['pk']
+        print(pk)
+        Contact.objects.filter(pk=pk).delete();
+
+        print(Contact.objects.all())
+
+        context = self.get_context_data(**kwargs)
+        return self.render_to_response(context)
+
+class EditContactView(UpdateView):
```

Comment:
Actually, you should create a brand new mixin for these two views (EditContactView and NewContactView) that inherits from `LoginRequiredMixin` AND also checks for a new "Can modify contacts" permission.